File: cpp/09.cpp
#include <iostream>
#include <thread>
#include <chrono>

void cpuIntensiveTask() {
    // Perform CPU-intensive task (example)
    for (int i = 0; i < 100000000; ++i) {
        double result = i * i;
    }
}

int main() {
    // Define the number of threads
    const int NUM_THREADS = 4;

    // Start the timer
    auto start = std::chrono::high_resolution_clock::now();

    // Create and start multiple threads
    std::thread threads[NUM_THREADS];
    for (int i = 0; i < NUM_THREADS; ++i) {
        threads[i] = std::thread(cpuIntensiveTask);
    }

    // Join all threads to the main thread
    for (int i = 0; i < NUM_THREADS; ++i) {
        threads[i].join();
    }

    // Stop the timer
    auto end = std::chrono::high_resolution_clock::now();
    std::chrono::duration<double> duration = end - start;

    // Display the time taken
    std::cout << "Time taken: " << duration.count() << " seconds" << std::endl;

    return 0;
}

File: python/09.py
import threading
import time

def cpu_intensive_task():
    # Perform CPU-intensive task (example)
    for i in range(100000000):
        result = i * i

def main():
    # Define the number of threads
    NUM_THREADS = 4

    # Start the timer
    start = time.time()

    # Create and start multiple threads
    threads = []
    for _ in range(NUM_THREADS):
        thread = threading.Thread(target=cpu_intensive_task)
        thread.start()
        threads.append(thread)

    # Join all threads to the main thread
    for thread in threads:
        thread.join()

    # Stop the timer
    end = time.time()
    duration = end - start

    # Display the time taken
    print("Time taken:", duration, "seconds")

if __name__ == "__main__":
    main()
